lib/CArtHandler.cpp

```diff
@@ -167,7 +167,7 @@ bool CArtifact::isBig() const
 
 bool CArtifact::isTradable() const
 {
-	switch(id)
+	switch(id.getNum())
```

Comment:
id of the same type as the switch-case (ArtifactID). Why do additional type conversion?

Reply:
Ah, should be toEnum(). Will change.

But no. Additional type conversion is needed.

ArtifactID is class. Weird class that sometimes acts like an integer, but still a class.
ArtifactID::Type is enumeration (and this is what should be used for `switch`)